openedx/core/djangoapps/credit/utils.py

```diff
@@ -0,0 +1,35 @@
+"""
+Utilities for the credit app.
+"""
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.django import modulestore
+
+
+def get_course_blocks(course_key, category):
+    """
+    Retrieve all XBlocks in the course for a particular category.
+
+    Returns only XBlocks that are published and haven't been deleted.
+    """
+    return [
+        block for block in modulestore().get_items(
+            course_key,
+            qualifiers={"category": category},
+            revision=ModuleStoreEnum.RevisionOption.published_only,
+        )
+        if _is_in_course_tree(block)
+    ]
```

Comment:
I'd prefer to keep it isolated to the credit app rather than to build it into the modulestore APIs.  The only reason we're doing this is because of [PLAT-799](https://openedx.atlassian.net/browse/PLAT-799) -- once that gets fixed, we can just use `get_items()`.  I'll add a comment to make this clear.